pkg/instancetype/instancetype.go

```diff
@@ -76,6 +80,49 @@ type InstancetypeMethods struct {
 
 var _ Methods = &InstancetypeMethods{}
 
+func (m *InstancetypeMethods) Expand(vm *virtv1.VirtualMachine, clusterConfig *virtconfig.ClusterConfig) (*virtv1.VirtualMachine, error) {
+	if vm.Spec.Instancetype == nil && vm.Spec.Preference == nil {
+		return vm, nil
+	}
+
+	instancetypeSpec, err := m.FindInstancetypeSpec(vm)
+	if err != nil {
+		return nil, err
+	}
+	preferenceSpec, err := m.FindPreferenceSpec(vm)
+	if err != nil {
+		return nil, err
+	}
+	expandedVM := vm.DeepCopy()
+
+	utils.SetDefaultVolumeDisk(&expandedVM.Spec.Template.Spec)
+
+	if err := vmispec.SetDefaultNetworkInterface(clusterConfig, &expandedVM.Spec.Template.Spec); err != nil {
+		return nil, err
+	}
+
+	conflicts := m.ApplyToVmi(
+		k8sfield.NewPath("spec", "template", "spec"),
+		instancetypeSpec, preferenceSpec,
+		&expandedVM.Spec.Template.Spec,
+		&expandedVM.Spec.Template.ObjectMeta,
+	)
+	if len(conflicts) > 0 {
```

Comment:
should we log these conflicts?

Reply:
Yeah I can add that now, I didn't initially as we shouldn't see conflicts here given the check in the admission webhook. That's something we've since talked about moving into the controller anyway so I may as well handle it now.